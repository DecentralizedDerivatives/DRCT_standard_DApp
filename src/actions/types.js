export const SET_USER_ACCOUNT = 'SET_USER_ACCOUNT';
export const SET_USER_BALANCE = 'SET_USER_BALANCE';
export const SET_USER_TRANSACTIONS = 'SET_USER_TRANSACTIONS';
export const SET_USER_POSITIONS = 'SET_USER_POSITIONS';
export const SET_CONTRACT_DETAILS = 'SET_CONTRACT_DETAILS';
export const SET_ORDER_DETAILS = 'SET_ORDER_DETAILS';
export const SET_SELECTED_TOKEN = 'SET_SELECTED_TOKEN';

export const SET_IS_CONNECTED = 'SET_IS_CONNECTED';
export const TX_PROCESSING = 'TEX_PROCESSING';
export const SET_TX_RECEIPT = 'SET_TX_RECEIPT';
export const RESET_TX_STATE = 'RESET_TX_STATE';
export const SET_PROCESSING_ERROR = 'SET_PROCESSING_ERROR';
export const CLEAR_PROCESSING_ERROR = 'CLEAR_PROCESSING_ERROR';
