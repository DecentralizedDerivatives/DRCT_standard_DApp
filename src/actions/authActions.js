import { web3 } from '../ethereum';

import { GET_ERRORS, SET_USER_CONNECTED } from './types';

// Check connection status 
export const checkConnection = () => dispatch => {
  try {
    const accounts = await web3.eth.getAccounts();
    const network = await web3.eth.net.getId();
    
    dispatch({
      type: SET_USER_CONNECTED,
      payload: (accounts.length && network === 4)  
      })
    }
  } catch(err) {
    dispatch({
      type: GET_ERRORS,
      payload: err.response.data
    })
  }
}
