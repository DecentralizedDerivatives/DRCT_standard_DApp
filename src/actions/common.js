import { Oracle } from '../ethereum';

const moment = require('moment');

export const getStartDatePrice = async (oracleAddress, startDate) => {
  let dt = startDate ? moment(startDate, 'MM/DD/YYYY').utc().startOf('day') : null;
  let date = dt.format('x') / 1000
  const oracle = await Oracle.at(oracleAddress)
  var data = await oracle.retrieveData(date)
  if (!data || !data.c || !data.c[0]) { return 0 }
  let val = parseFloat('0.00' + data.c[0].toString())
  // console.log('val', val)
  let multiplier = '1e' + (data.e).toString()
  let adjustedValue = val * (Number(multiplier))
  // console.log('adjustedValue', adjustedValue)
  return adjustedValue
}
