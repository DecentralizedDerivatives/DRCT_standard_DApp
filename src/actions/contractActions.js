import { Factory, Exchange, DRCT, Oracle } from '../ethereum';
import {
  SET_CONTRACT_DETAILS,
  SET_CONTRACT_OPEN_DATES,
  SET_CONTRACT_START_PRICE,
  SET_ORDERBOOK,
  SET_FETCH_IN_PROGRESS,
  REMOVE_FETCH_IN_PROGRESS,
  SET_FETCHING_ERROR,
  SET_RECENT_TRADES
} from './types';

import FactoryProvider from '../factoryProvider';
const moment = require('moment');

export const getContractDetails = (symbol) => async dispatch => {
  try {
    const provider = FactoryProvider.getFromSymbol(symbol);
    // const factory = await Factory.at(provider && provider.address ? provider.address : '');
    // const response = await factory.getVariables();
    const details = {
      contractAddress: provider.address,
      contractDuration: provider.duration,
      contractMultiplier: provider.multiplier,
      oracleAddress: provider.oracle
      // contractAddress: response[0],
      // contractDuration: response[1].c[0],
      // contractMultiplier: response[2].c[0],
      // oracleAddress: response[3]
    };

    dispatch({
      type: SET_CONTRACT_DETAILS,
      payload: details
    });
    return details;
  } catch (err) {
    dispatch({
      type: SET_FETCHING_ERROR,
      payload: err.message.split('\n')[0]
    });
  }
};

export const getStartDatePrice = (oracleAddress, startDate) => async dispatch => {
  try {
    const oracle = await Oracle.at(oracleAddress)
    var data = await oracle.retrieveData(startDate);
    dispatch({
      type: SET_CONTRACT_START_PRICE,
      payload: data.c[0]
    });
  } catch (err) {
    dispatch({
      type: SET_FETCHING_ERROR,
      payload: err.message.split('\n')[0]
    });
  }
}
export const getOrderBook = (isSilent) => async dispatch => {
  try {
    if (!isSilent) { dispatch({ type: SET_FETCH_IN_PROGRESS, payload: SET_ORDERBOOK }); };
    var staticAddresses = FactoryProvider.getStaticAddresses();
    const exchange = await Exchange.at(staticAddresses.exchange);
    let numBooks = await exchange.getBookCount();
    var factories = FactoryProvider.factories();
    let _allrows = [];
    for (let i = 0; i < numBooks; i++) {
      let book = await exchange.openBooks(i);
      // console.log('book', book)
      for (var p = 0; p < factories.length; p++) {
        const factory = await Factory.at(factories[p].address);
        // console.log('factory', factories[p].symbol);
        let tokenDate = await factory.token_dates.call(book);
        // console.log('tokenDate', tokenDate)
        if (tokenDate.c[0] === 0) { continue }
        let orders = await exchange.getOrders(book);
        // console.log('orders', orders)
        for (let j = 0; j < orders.length; j++) {
          if (orders[j].c[0] > 0) {
            let order = await exchange.getOrder(orders[j].c[0]);
            // console.log('order', order)
            let tokenType = (await factory.getTokenType(order[3])).c[0];
            let date = new Date(tokenDate.c[0] * 1000);
            let startDate = new Date();
            startDate.setDate(date.getDate() + 1);
            var todayMinusSixDays = new Date();
            todayMinusSixDays.setDate(todayMinusSixDays.getDate() - 6);
            if (date > todayMinusSixDays) {
              let orderDate = startDate.getUTCMonth() + 1 + '/' +
                startDate.getUTCDate() + '/' + startDate.getUTCFullYear();
              var precisePrice = parseFloat(order[1].c[0]/10000).toFixed(5);
              _allrows.push({
                orderId: orders[j].c[0].toString(),
                address: order[3],
                price: precisePrice,
                quantity: order[2].c[0].toString(),
                date: orderDate.toString(),
                symbol: factories[p].symbol,
                tokenType: tokenType === 1 ? 'Short' : 'Long'
               });
            }
          }
        }
      }
      _allrows.sort(function (a, b) {
        return a.orderId - b.orderId;
      });
    }
    dispatch({
      type: SET_ORDERBOOK,
      payload: _allrows
    });
    dispatch({ type: REMOVE_FETCH_IN_PROGRESS, payload: SET_ORDERBOOK });
  } catch (err) {
    dispatch({
      type: SET_FETCHING_ERROR,
      payload: 'Order Book: ' + err.message.split('\n')[0]
    });
  }
};

export const getRecentTrades = (isSilent) => async dispatch => {
  try {
    if (!isSilent) { dispatch({ type: SET_FETCH_IN_PROGRESS, payload: SET_RECENT_TRADES }); };
    var staticAddresses = FactoryProvider.getStaticAddresses();
    const exchange = await Exchange.at(staticAddresses.exchange);

    let transferEvent = await exchange.Sale(
      {},
      { fromBlock: 0, toBlock: 'latest' }
    );

    transferEvent.get(async function (err, events) {
      // console.log('events', events);
      var trades = [];
      if (events.length > 0) {
        for (let i = events.length - 1; i >= Math.max(events.length - 10, 0); i--) {
          var token = events[i].args['_token'].toString();
          // console.log('token', token);
          var drct = DRCT.at(token);
          // console.log('drct', drct);
          var factoryAddress = await drct.getFactoryAddress();
          const factory = await Factory.at(factoryAddress);
          let tokenType = (await factory.getTokenType(token)).c[0];
          // console.log('factoryAddress', factoryAddress);
          var provider = FactoryProvider.getFromAddress(factoryAddress);
          var precisePrice = parseFloat(events[i].args['_price']/1e18).toFixed(5);
          trades.push({
            address: token,
            volume: events[i].args['_amount'].toString(),
            price: precisePrice,
            contractDuration: provider && provider.duration ? provider.duration : 0,
            contractMultiplier: provider && provider.multiplier ? provider.multiplier : 0,
            symbol: provider && provider.symbol ? provider.symbol : '??',
            tokenType: tokenType === 1 ? 'Short' : 'Long'
          });
        }
      }
      dispatch({
        type: SET_RECENT_TRADES,
        payload: trades
      });
      dispatch({ type: REMOVE_FETCH_IN_PROGRESS, payload: SET_RECENT_TRADES });
    });
  } catch (err) {
    dispatch({
      type: SET_FETCHING_ERROR,
      payload: 'Recent Trades: ' + err.message.split('\n')[0]
    });
  }
};

export const getContractOpenDates = (currencyAddress) => async dispatch => {
  try {
    const factory = await Factory.at(currencyAddress);
    // console.log(factory);
    let openDates = {};
    const numDates = await factory.getDateCount();

    for (let i = 0; i < numDates; i++) {
      const startDate = (await factory.startDates.call(i)).c[0];
      let formattedDate = new Date(startDate * 1000);
      var momentDate = moment(formattedDate);
      var pastDate = moment().subtract(5, 'days').startOf('day');
      if (momentDate.isBefore(pastDate)) { continue }
      formattedDate = formattedDate.getUTCMonth() + 1 + '/' +
        formattedDate.getUTCDate() + '/' +
        formattedDate.getUTCFullYear();
      openDates[startDate] = formattedDate;
    }
    dispatch({
      type: SET_CONTRACT_OPEN_DATES,
      payload: openDates
    });
  } catch (err) {
    dispatch({
      type: SET_FETCHING_ERROR,
      payload: 'Contract Open Dates: ' + err.message.split('\n')[0]
    });
  }
};
