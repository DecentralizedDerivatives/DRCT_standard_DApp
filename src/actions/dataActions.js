import api from '../api';
import { SET_PRICECHART_DATA, SET_FETCHING_ERROR } from './types';

export const getPriceChartData = type => async dispatch => {
  try {
    const data = await api[type].get();
      
    dispatch({
      type: SET_PRICECHART_DATA,
      payload: data 
    });      

  } catch (err) {
    dispatch({
      type: SET_FETCHING_ERROR,
      payload: 'PriceChart: ' + err.message.split('\n')[0]
    });
  }
};
