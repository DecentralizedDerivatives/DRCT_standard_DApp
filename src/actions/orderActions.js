import { Factory, Exchange, DRCT, UserContract } from '../ethereum';
import {
  SET_ORDER_DETAILS,
  SET_BUY_ORDER_RECEIPT,
  SET_BUY_ORDER_ERROR,
  SET_UNLIST_ORDER_RECEIPT,
  SET_UNLIST_ORDER_ERROR,
  SET_CONTRACT_CREATED,
  SET_CONTRACT_ERROR,
  SET_CONTRACT_FUNDED,
  SET_SEND_FUNDS_ERROR,
  SET_LIST_ORDER,
  SET_LIST_ORDER_ERROR,
  SET_LIST_ORDER_APPROVED,
  SET_LIST_ORDER_APPROVE_ERROR,
  SET_PROCESSING,
  SET_FETCHING_ERROR
} from './types';

import FactoryProvider from '../factoryProvider';

//TODO: This function makes no sense.  Why return an array of strings??
//Should be a single object with order details.  Right??
export const getOrderDetails = orderId => async dispatch => {
  try {
    if (!orderId) {
      return dispatch({type: SET_ORDER_DETAILS, payload: ''});
    }
    var factories = FactoryProvider.factories();
    var staticAddresses = FactoryProvider.getStaticAddresses();
    const exchange = await Exchange.at(staticAddresses.exchange);
    let order = await exchange.getOrder(orderId);
    let _allrows = [];
    for (var i = 0; i < factories.length; i++) {
      const factory = await Factory.at(factories[i].address);
      var orderDetail = await getOrderDetailForFactory(factory, order, orderId);
      _allrows.push(orderDetail);
    }
    dispatch({
      type: SET_ORDER_DETAILS,
      payload: _allrows
    });
  } catch (err) {
    dispatch({
      type: SET_FETCHING_ERROR,
      payload: 'Get Order Details: ' + err.message.split('\n')[0]
    });
  }
};

const getOrderDetailForFactory = async (factory, order, orderId) => {
  let display = [];
  let date = await factory.token_dates.call(order[3]);
  date = new Date(date * 1000);
  date =
    date.getMonth() + 1 + '/' + date.getDate() + '/' + date.getFullYear();
  display =
    orderId.toString() +
    '(' +
    order[3] +
    order[1].c[0].toString() +
    '/' +
    order[2].c[0].toString() +
    '/' +
    date.toString() +
    ')';
  return display;
};

export const sendBuyOrder = (orderId, account) => async dispatch => {
  dispatch(setProcessing(true));
  dispatch({type: SET_BUY_ORDER_ERROR, payload: null });
  try {
    orderId = parseInt(orderId, 10);
    var staticAddresses = FactoryProvider.getStaticAddresses();
    const exchange = await Exchange.at(staticAddresses.exchange);
    const order = await exchange.getOrder(orderId);
    const _value = order[1];
    const response = await exchange.buy(orderId, {
      from: account,
      gas: 400000,
      value: _value
    });
    // console.log('RESPONSE', response)
    dispatch({
      type: SET_BUY_ORDER_RECEIPT,
      payload: {
        id: response.tx,
        orderId: orderId
      }
    });
  } catch (err) {
    // console.log('err', err.message)
    dispatch({
      type: SET_BUY_ORDER_ERROR,
      payload: err.message.split('\n')[0]
    });
  }
  dispatch(setProcessing(false));
};

export const sendUnlistOrder = (orderId, account) => async dispatch => {
  dispatch(setProcessing(true));

  try {
    var staticAddresses = FactoryProvider.getStaticAddresses();
    const exchange = await Exchange.at(staticAddresses.exchange);
    const response = await exchange.unlist(orderId, {
      from: account,
      gas: 200000
    });

    dispatch({
      type: SET_UNLIST_ORDER_RECEIPT,
      payload: {
        id: response.tx,
        orderId: orderId
      }
    });
  } catch (err) {
    dispatch({
      type: SET_UNLIST_ORDER_ERROR,
      payload: err.message.split('\n')[0]
    });
  }

  dispatch(setProcessing(false));
};

export const sendListOrder = (formValues, account) => async dispatch => {
  dispatch(setProcessing(true));

  let { token, tokenAmount, price } = formValues;

  try {
    var staticAddresses = FactoryProvider.getStaticAddresses();
    const exchange = await Exchange.at(staticAddresses.exchange);

    const response = await exchange.list(token, tokenAmount, price * 1e18, {
      from: account,
      gas: 400000
    });

    dispatch({
      type: SET_LIST_ORDER,
      payload: {
        id: response.tx,
        price
      }
    });
  } catch (err) {
    dispatch({
      type: SET_LIST_ORDER_ERROR,
      payload: err.message.split('\n')[0]
    });
  }

  dispatch(setProcessing(false));
};

export const sendApproveOrder = (approveDetails, account) => async dispatch => {
  dispatch(setProcessing(true));

  let { token, tokenAmount } = approveDetails;

  try {
    var staticAddresses = FactoryProvider.getStaticAddresses();
    const exchange = await Exchange.at(staticAddresses.exchange);
    const drct = await DRCT.at(token);
    const allowance = await drct.allowance(account, exchange.address)
    const approvedAmount = allowance && allowance.c && allowance.c.length > 0 ? allowance.c[0] : 0
    let listedAmount = tokenAmount
    if (Number(tokenAmount) < Number(approvedAmount)) {
      const ordersList = await exchange.getUserOrders(account)
      for (let ord of ordersList) {
        let order = await exchange.getOrder(ord.c[0])
        if (order[3] === token) {
          listedAmount += order[2].c[0]
        }
      }
    }
    if (Number(listedAmount) > Number(approvedAmount)) {
      await drct.approve(exchange.address, tokenAmount, {
        from: account,
        gas: 100000
      });
    }
    dispatch({
      type: SET_LIST_ORDER_APPROVED,
      payload: {
        token,
        tokenAmount,
        approved: true
        // approveTx: response.tx
      }
    });
  } catch (err) {
    console.log('APPROVE ERROR', err)
    dispatch({
      type: SET_LIST_ORDER_APPROVE_ERROR,
      payload: err.message.split('\n')[0]
    });
  }

  dispatch(setProcessing(false));
};

export const sendCreateContractOrder = (
  formValues,
  account
) => async dispatch => {
  dispatch(setProcessing(true));
  
  const { duration, currency, startDate } = formValues;
  var amount = formValues.amount / 2;
  try {
    const factory = await Factory.at(currency);

    let userAddress = await factory.user_contract.call();
    const userContract = await UserContract.at(userAddress);

    let fundingAmount = 1e18 * amount;
    console.log('Variables',startDate,fundingAmount)
    const response = await userContract.Initiate(startDate, fundingAmount, {
      from: account,
      gas: 1100000,
      value: fundingAmount * 2
    });

    dispatch({
      type: SET_CONTRACT_FUNDED,
      payload: response.tx
    });
  } catch (err) {
    dispatch({
      type: SET_SEND_FUNDS_ERROR,
      payload: err.message.split('\n')[0]
    });
  }
  dispatch(setProcessing(false));
};

export const setProcessing = status => {
  return {
    type: SET_PROCESSING,
    payload: status
  };
};
