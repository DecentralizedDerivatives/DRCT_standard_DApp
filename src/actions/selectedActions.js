import { SET_SELECTED_TOKEN, CLEAR_SELECTED_TOKEN } from './types';

export const setSelectedToken = token => dispatch => {
  dispatch({
    type: SET_SELECTED_TOKEN,
    payload: token
  });
};

export const clearSelectedToken = token => dispatch => {
  dispatch({
    type: CLEAR_SELECTED_TOKEN,
    payload: ''
  });
};
