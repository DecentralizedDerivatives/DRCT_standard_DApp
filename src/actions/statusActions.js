import { web3 } from '../ethereum';

import {
  SET_FETCHING_ERROR,
  SET_CONNECTION_STATUS,
  SHOW_CONNECTION_MODAL,
  SET_PROCESSING
} from './types';

// Check connection status
export const checkUserConnection = () => async dispatch => {
  try {
    const accounts = await web3.eth.getAccounts();
    const network = await web3.eth.net.getId();
    console.log("NETWORK",network);
    dispatch({
      type: SET_CONNECTION_STATUS,
      payload: {
        metamask: Boolean(accounts.length),
        network: network
      }
    });
  } catch (err) {
    dispatch({
      type: SET_FETCHING_ERROR,
      payload: err.message.split('\n')[0]
    });
  }
};

// Set connection modal status
export const showConnectionModal = status => {
  return {
    type: SHOW_CONNECTION_MODAL,
    payload: status
  };
};

export const setProcessing = status => {
  return {
    type: SET_PROCESSING,
    payload: status
  };
};
