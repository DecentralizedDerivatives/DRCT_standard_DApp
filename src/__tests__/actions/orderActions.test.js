import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {
  SET_ORDER_DETAILS,
  SET_BUY_ORDER_RECEIPT,
  SET_BUY_ORDER_ERROR,
  SET_UNLIST_ORDER_RECEIPT,
  SET_UNLIST_ORDER_ERROR,
  SET_CONTRACT_CREATED,
  SET_CONTRACT_ERROR,
  SET_CONTRACT_FUNDED,
  SET_SEND_FUNDS_ERROR,
  SET_LIST_ORDER,
  SET_LIST_ORDER_ERROR,
  SET_LIST_ORDER_APPROVED,
  SET_LIST_ORDER_APPROVE_ERROR,
  SET_PROCESSING,
  SET_FETCHING_ERROR
} from '../../actions/types';

import * as orderActions from '../../actions/orderActions';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const store = mockStore({});

describe('orderActions', () => {
  afterEach(() => {
    store.clearActions();
  });

  describe('getOrderDetails', () => {
    it('Handles empty order id', () => {
      const store = mockStore({});
      store.dispatch(orderActions.getOrderDetails());
      const actions = store.getActions();
      const expectedActions = {
        type: SET_ORDER_DETAILS,
        payload: ''
      };
      expect(actions).toEqual([expectedActions]);
    });
    // it('Handles fake order id', () => {
    //   const store = mockStore({});
    //   store.dispatch(orderActions.getOrderDetails(1));
    //   const actions = store.getActions();
    //   const expectedActions = {
    //     type: SET_FETCHING_ERROR
    //   };
    //   expect(actions).toEqual([expectedActions]);
    // });
  });

  // describe('showConnectionModal', () => {
  //   it('should dispatch correct action', () => {
  //     const store = mockStore({});
  //     store.dispatch(statusActions.showConnectionModal(true));
  //
  //     const actions = store.getActions();
  //     const expectedActions = {
  //       type: 'SHOW_CONNECTION_MODAL',
  //       payload: true
  //     };
  //     expect(actions).toEqual([expectedActions]);
  //   });
  // });
  //
  // describe('checkUserConnection', () => {
  //   const defaultStatusState = {
  //     connectStatus: {
  //       metamask: false,
  //       network: 0,
  //       verified: false,
  //       whiteListed: false
  //     }
  //   };
  //
  //   it('should dispatch correct action', done => {
  //     const store = mockStore(defaultStatusState);
  //     store.dispatch(statusActions.checkUserConnection()).then(() => {
  //       const actions = store.getActions();
  //       const expectedActions = {
  //         type: 'SET_CONNECTION_STATUS',
  //         payload: {
  //           metamask: false,
  //           network: 0,
  //           verified: true,
  //           whiteListed: false
  //         }
  //       };
  //       expect(actions[0]).toEqual(expectedActions);
  //     });
  //     done();
  //   });
  // });
});
