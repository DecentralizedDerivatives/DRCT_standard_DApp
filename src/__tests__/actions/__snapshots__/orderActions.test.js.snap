// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orderActions getOrderDetails error 1`] = `
Array [
  Object {
    "payload": "Get Order Details: Cannot read property '3' of undefined",
    "type": "SET_FETCHING_ERROR",
  },
]
`;

exports[`orderActions getOrderDetails with id 1`] = `
Array [
  Object {
    "payload": Array [
      "0x000...(0x000...20/30/1/1/2018)",
    ],
    "type": "SET_ORDER_DETAILS",
  },
]
`;

exports[`orderActions getOrderDetails without id 1`] = `
Array [
  Object {
    "payload": "",
    "type": "SET_ORDER_DETAILS",
  },
]
`;

exports[`orderActions sendApproveOrder 1`] = `
Array [
  Object {
    "payload": true,
    "type": "SET_PROCESSING",
  },
  Object {
    "payload": Object {
      "approved": true,
      "token": "0x000...",
      "tokenAmount": 1,
    },
    "type": "SET_LIST_ORDER_APPROVED",
  },
  Object {
    "payload": false,
    "type": "SET_PROCESSING",
  },
]
`;

exports[`orderActions sendApproveOrder error 1`] = `
Array [
  Object {
    "payload": true,
    "type": "SET_PROCESSING",
  },
  Object {
    "payload": "error",
    "type": "SET_LIST_ORDER_APPROVE_ERROR",
  },
  Object {
    "payload": false,
    "type": "SET_PROCESSING",
  },
]
`;

exports[`orderActions sendApproveOrder no allowance received 1`] = `
Array [
  Object {
    "payload": true,
    "type": "SET_PROCESSING",
  },
  Object {
    "payload": Object {
      "approved": true,
      "token": "0x000...",
      "tokenAmount": 1,
    },
    "type": "SET_LIST_ORDER_APPROVED",
  },
  Object {
    "payload": false,
    "type": "SET_PROCESSING",
  },
]
`;

exports[`orderActions sendApproveOrder, no token in orders 1`] = `
Array [
  Object {
    "payload": true,
    "type": "SET_PROCESSING",
  },
  Object {
    "payload": Object {
      "approved": true,
      "token": "0x111...",
      "tokenAmount": 1,
    },
    "type": "SET_LIST_ORDER_APPROVED",
  },
  Object {
    "payload": false,
    "type": "SET_PROCESSING",
  },
]
`;

exports[`orderActions sendApproveOrder, token amount is more than approved 1`] = `
Array [
  Object {
    "payload": true,
    "type": "SET_PROCESSING",
  },
  Object {
    "payload": Object {
      "approved": true,
      "token": "0x000...",
      "tokenAmount": 100,
    },
    "type": "SET_LIST_ORDER_APPROVED",
  },
  Object {
    "payload": false,
    "type": "SET_PROCESSING",
  },
]
`;

exports[`orderActions sendBuyOrder 1`] = `
Array [
  Object {
    "payload": true,
    "type": "SET_PROCESSING",
  },
  Object {
    "payload": null,
    "type": "SET_BUY_ORDER_ERROR",
  },
  Object {
    "payload": Object {
      "id": "0x000...",
      "orderId": NaN,
    },
    "type": "SET_BUY_ORDER_RECEIPT",
  },
  Object {
    "payload": false,
    "type": "SET_PROCESSING",
  },
]
`;

exports[`orderActions sendBuyOrder error 1`] = `
Array [
  Object {
    "payload": true,
    "type": "SET_PROCESSING",
  },
  Object {
    "payload": null,
    "type": "SET_BUY_ORDER_ERROR",
  },
  Object {
    "payload": "Cannot read property 'tx' of undefined",
    "type": "SET_BUY_ORDER_ERROR",
  },
  Object {
    "payload": false,
    "type": "SET_PROCESSING",
  },
]
`;

exports[`orderActions sendCreateContractOrder 1`] = `
Array [
  Object {
    "payload": true,
    "type": "SET_PROCESSING",
  },
  Object {
    "payload": "0x000...",
    "type": "SET_CONTRACT_FUNDED",
  },
  Object {
    "payload": false,
    "type": "SET_PROCESSING",
  },
]
`;

exports[`orderActions sendCreateContractOrder error 1`] = `
Array [
  Object {
    "payload": true,
    "type": "SET_PROCESSING",
  },
  Object {
    "payload": "Cannot read property 'tx' of undefined",
    "type": "SET_SEND_FUNDS_ERROR",
  },
  Object {
    "payload": false,
    "type": "SET_PROCESSING",
  },
]
`;

exports[`orderActions sendListOrder 1`] = `
Array [
  Object {
    "payload": true,
    "type": "SET_PROCESSING",
  },
  Object {
    "payload": Object {
      "id": "0x000...",
      "price": 1,
    },
    "type": "SET_LIST_ORDER",
  },
  Object {
    "payload": false,
    "type": "SET_PROCESSING",
  },
]
`;

exports[`orderActions sendListOrder error 1`] = `
Array [
  Object {
    "payload": true,
    "type": "SET_PROCESSING",
  },
  Object {
    "payload": "Cannot read property 'tx' of undefined",
    "type": "SET_LIST_ORDER_ERROR",
  },
  Object {
    "payload": false,
    "type": "SET_PROCESSING",
  },
]
`;

exports[`orderActions sendUnlistOrder 1`] = `
Array [
  Object {
    "payload": true,
    "type": "SET_PROCESSING",
  },
  Object {
    "payload": Object {
      "id": "0x000...",
      "orderId": undefined,
    },
    "type": "SET_UNLIST_ORDER_RECEIPT",
  },
  Object {
    "payload": false,
    "type": "SET_PROCESSING",
  },
]
`;

exports[`orderActions sendUnlistOrder error 1`] = `
Array [
  Object {
    "payload": true,
    "type": "SET_PROCESSING",
  },
  Object {
    "payload": "Cannot read property 'tx' of undefined",
    "type": "SET_UNLIST_ORDER_ERROR",
  },
  Object {
    "payload": false,
    "type": "SET_PROCESSING",
  },
]
`;

exports[`orderActions setProcessing 1`] = `
Array [
  Object {
    "payload": undefined,
    "type": "SET_PROCESSING",
  },
]
`;
